src/reference/asciidoc/codec.adoc

```diff
@@ -0,0 +1,122 @@
+[[codec]]
+=== Codec
+
+==== Introduction
+
+Spring Integration _version 4.2_ introduces the `Codec` abstraction.
+Codecs are used to encode/decode objects to/from byte[].
+They are an alternative to Java Serialization.
+One advantage is, typically, objects do not have to implement Serializable
+One implementation, using https://github.com/EsotericSoftware/kryo[Kryo] for serialization is provided but you
+can provide your own implementation for use in any of these components:
+
+* EncodingPayloadTransformer
+* DecodingTransformer
+* CodecMessageConverter
+
+See their JavaDocs for more information.
+
+==== EncodingPayloadTransformer
+
+This transformer encodes the payload to a `byte[]` using the codec.
+It does not affect message headers.
+
+==== DecodingTransformer
+
+This transformer decodes a `byte[]` using the codec; it needs to be configured with the Class to which the object
+should be decoded (or an expression that resolves to a Class).
+If the resulting object is a `Message<?>`, inbound headers will not be retained.
+
+==== CodecMessageConverter
+
+Certain endpoints (e.g. TCP, Redis) have no concept of message headers; they support the use of a
+`MessageConverter` and the `CodecMessageConverter` can be used to convert a message to/from a `byte[]` for
+transmission.
+
+==== Kryo
+
+Currently, this is the only implementation of `Codec`.
+There are two `Codec`s - `PojoCodec` which can be used in the transformers and `MessageCodec` which can be used
+in the `CodecMessageConverter`.
+
+===== Customizing Kryo
```

Comment:
This one is present as an item in the contents, because it has 5 `=` in the beginning, but all the rest like `Using a Custom Kryo Serializer` and so on, isn't because of 6 for those `=` symbols.
Is that intentional?
Thanks